### PR TITLE
NH-10686 fix flaky tests

### PR DESCRIPTION
**Pull request by prettycoder**
In order of appearance:
- rename add_kvs to add_traceinfo
- bunny: use incoming tracestate for continuing trace, keep SourceTraces as a second option
- fix graphql bug and check presence
- add SW_APM_TRACE_STARTED entry to rack env to keep the context of a started request
- remove using existing Context from start_trace and log_start, to be cont in NH-11132
- **most of the changes are about resetting context in tests and printing traces if assertion fails**

This will not be released as is:
- some tests are skipped and will be addressed with NH-11132
- some comments about flaky test will be removed in the future

### CODE REVIEW

File: lib/solarwinds_apm/inst/bunny-consumer.rb
Comment (ronilan) [inline lib/solarwinds_apm/inst/bunny-consumer.rb:71]: seen this....

File: lib/solarwinds_apm/inst/rack.rb
Comment (ronilan) [inline lib/solarwinds_apm/inst/rack.rb:166]: If I understand correctly - this is similar to the way the async context is kept in node, and the question is, why wasn't it needed before? And if it wasn't what is it in NH that requires this feature?

**Review by ronilan — APPROVED** [review]
Left one question inline.

Also, since the PR is already big and touches the api, instrumented gems and tests in one go - from where I look at things - adding NH-11132 tasks before merge may actually make things easier comprehend.

Your call (obviously).

File: lib/solarwinds_apm/inst/rack.rb
Comment (prettycoder) [reply lib/solarwinds_apm/inst/rack.rb:166]: There are cases where rack is re-entered during the same request. In the past the context was kept and continued.
But if something breaks and context is around for a different reason, we will still continue it. This is a safer approach, because we know when to continue and when to clean up. 
It was needed long before ... cleaning up loose ends

**Comment by prettycoder** [comment]
I'll leave this PR open and see which PR will be easier to look at after the next round of refactorings

I was hoping for more of a clean line between the 2 ... running out of time

**Pull request merged**
